management-interface.go

```diff
@@ -49,7 +52,11 @@ var tmpl *template.Template
 // This does some initialisation.  It parses our html templates up front and
 // finds the location where this executable was started.
 func init() {
+
+	setDeviceName()
+
 	tmpl = template.New("")
+	tmpl.Funcs(template.FuncMap{"DeviceName": getDeviceName})
```

Comment:
Using a template function is nice workaround.

You can avoid the need for the `deviceName` module level variable and the current getDeviceName function by using a closure. Have setDeviceName just return the device name (and perhaps call it getDeviceName!) and then do this here:

```go
deviceName := getDeviceName()
tmpl.Funcs(template.FuncMap{"DeviceName": func() { return deviceName }})
```